blog/2023-10-10-weaviate-1-22-release/index.mdx

```diff
@@ -0,0 +1,253 @@
+        "child": {
+            "childname": "Nested child",
+            "granchild": "I'm doubly nested!"
+        }
+    }
+},
+```
+
+</details>
+
+More practically, this type of structure may be applied to capture addresses, product information, or any other nested data structure. Take a look at the following example, which shows a `Person` class with an `address` property that is an `object` datatype. The `address` property is able to store its own `nestedProperties` of `street` and `city`.
+
+```json
+{
+    "class": "Person",
+    "properties": [
+        {
+            "dataType": ["text"],
+            "name": "last_name",
+        },
+        {
+            "dataType": ["object"],
+            "name": "address",
+            "nestedProperties": [
+                {"dataType": ["text"], "name": "street"},
+                {"dataType": ["text"], "name": "city"}
+            ],
+        }
+    ],
+}
+```
+
+The user may then create an object of this class, such as:
+
+```json
+{
+    "last_name": "Franklin",
+    "address": {
+        "city": "London",
+        "street": "King Street"
+    }
+}
+```
+
+### Auto-schema support for nested properties
+
+Given that the `object` data types allow nested properties of arbitrary depth, you will be pleased to know that auto-schema supports nested properties. This means that you can create a class with nested properties without having to manually define the class schema.
+
+In other words - you can create an `object` property, and Weaviate will automatically create the nested properties for you at import time.
+
+As with before, you will get the most granular and exact control over the schema by defining it manually. However, auto-schema may save you a lot of time, for example if you have a large number of nested properties.
+
+### Future improvements
+
+Please keep in mind that this is the first iteration of nested JSON object storage in Weaviate, and we have further improvements planned.
+
+In the initial implementation, any properties with `object` datatype do not affect the object vector, or the inverted index. This behavior is similar to what would happen with a cross-reference property, where the cross-referenced object is not indexed.
+
+Future plans include the ability to index nested properties, for example to allow for filtering on nested properties and vectorization options.
+
+## Async vector indexing
+
+Weaviate now offers the option of asynchronous vector indexing to object creation. In turn, it enables you to optimize imports for speed where vector indexing is not required immediately.
+
+In the past, the HNSW index would be built synchronously with object creation. This meant that the import process would occur in lockstep with the indexing process. The benefit of the synchronous approach is that the user can perform a search immediately after import, as the data is already indexed. However, if your only goal was fastest data ingestion, this could slow down the import process.
```

Comment:
Consider something like:
 
In earlier versions of Weaviate, the HNSW index builds during object creation. Data import takes place in lockstep with the indexing process. The benefit of the synchronous approach is that you can perform a search immediately after import because the data is already indexed. However, synchronous index creation slows down the import process, slowdown is not ideal if your goal is fast data ingestion.